pkg/controller/operators/olm/operator.go

```diff
@@ -590,44 +602,54 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 			operatorGroup = operatorGroups[0]
 			logger = logger.WithField("opgroup", operatorGroup.GetName())
 
-			a.addOperatorGroupAnnotations(&out.ObjectMeta, operatorGroup, true)
-			_, err = a.client.OperatorsV1alpha1().ClusterServiceVersions(out.GetNamespace()).Update(out)
-			if err != nil {
-				logger.Error("error adding operatorgroup annotation")
-				syncError = err
+			if a.operatorGroupAnnotationsDiffer(&out.ObjectMeta, operatorGroup) {
```

Comment:
Checking for differences first and then set the annotations - before, the map referenced by the pointer was being changed and then checked against itself.